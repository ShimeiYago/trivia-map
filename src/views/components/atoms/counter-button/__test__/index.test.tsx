import { shallow, ShallowWrapper } from 'enzyme';
import { CounterButton, Props } from '..';

let wrapper: ShallowWrapper<Props, unknown, CounterButton>;

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    wrapper = shallow(<CounterButton />);
  });

  it('basic', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('with props', () => {
    wrapper.setProps({
      variant: 'async',
      ariaLabel: 'label',
      children: 'button',
      onClick: jest.fn(),
      disabled: true,
    });
    expect(wrapper).toMatchSnapshot();
  });
});
