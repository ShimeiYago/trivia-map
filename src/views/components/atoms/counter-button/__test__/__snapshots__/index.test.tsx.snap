// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `
<button
  className="button"
/>
`;

exports[`Shallow Snapshot Tests with props 1`] = `
<button
  aria-label="label"
  className="asyncButton"
  disabled={true}
  onClick={[MockFunction]}
>
  button
</button>
`;
